metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/ExecutionStatusAggregator.java

```diff
@@ -0,0 +1,84 @@
+package io.dockstore.metricsaggregator.helper;
+
+import static java.util.stream.Collectors.groupingBy;
+
+import io.dockstore.openapi.client.model.ExecutionStatusMetric;
+import io.dockstore.openapi.client.model.Metrics;
+import io.dockstore.openapi.client.model.RunExecution;
+import io.dockstore.openapi.client.model.RunExecution.ExecutionStatusEnum;
+import io.dockstore.openapi.client.model.TaskExecutions;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Aggregate Execution Status metrics by summing up the count of each Execution Status.
+ */
+public class ExecutionStatusAggregator implements RunExecutionAggregator<ExecutionStatusMetric, ExecutionStatusEnum> {
+
+    @Override
+    public ExecutionStatusMetric getMetricFromMetrics(Metrics metrics) {
+        return metrics.getExecutionStatusCount();
+    }
+
+    @Override
+    public ExecutionStatusEnum getMetricFromRunExecution(RunExecution runExecution) {
+        return runExecution.getExecutionStatus();
+    }
+
+    @Override
+    public Optional<RunExecution> getWorkflowExecutionFromTaskExecutions(TaskExecutions taskExecutionsForOneWorkflowRun) {
```

Comment:
This is the new function that aggregates tasks into a workflow execution for the execution status metric